src/dash-table/dash/DataTable.js

```diff
@@ -1,41 +1,15 @@
-import * as R from 'ramda';
-import React, { Component } from 'react';
+import React, { Component, Suspense, lazy } from 'react';
 import PropTypes from 'prop-types';
 
-import RealTable from 'dash-table/components/Table';
+const LazyDataTable = lazy(() => import(/* webpackChunkName: "dash-table", webpackMode: "$${{mode}}" */ 'dash-table/dash/LazyDataTable'));
 
-import Logger from 'core/Logger';
-
-import genRandomId from 'dash-table/utils/generate';
-import isValidProps from './validate';
-import Sanitizer from './Sanitizer';
-
-/**
- * Dash DataTable is an interactive table component designed for
- * designed for viewing, editing, and exploring large datasets.
- * DataTable is rendered with standard, semantic HTML <table/> markup,
- * which makes it accessible, responsive, and easy to style. This
- * component was written from scratch in React.js specifically for the
- * Dash community. Its API was designed to be ergonomic and its behavior
- * is completely customizable through its properties.
- */
 export default class DataTable extends Component {
-    constructor(props) {
-        super(props);
-        let id;
-        this.getId = () => (id = id || genRandomId('table-'));
-        this.sanitizer = new Sanitizer();
-    }
-
     render() {
-        if (!isValidProps(this.props)) {
-            return (<div>Invalid props combination</div>);
-        }
-
-        const sanitizedProps = this.sanitizer.sanitize(this.props);
-        return this.props.id ?
-            (<RealTable {...sanitizedProps} />) :
-            (<RealTable {...sanitizedProps} id={this.getId()} />);
+        return (<Suspense
+            fallback={null}
+        >
+            <LazyDataTable {...this.props} />
+        </Suspense>);
```

Comment:
Lazy load the table implementation until it's actually required. This could be a case for `/* webpackPreload: true */` instead of maximum laziness.

The additional code will not be needed to begin rendering but will be made available soon after.

Reply:
The props are defined on the wrapper and reused by the lazy component for devtools to work as expected